packages/lab/src/badge/Badge.tsx

```diff
@@ -60,31 +34,25 @@ export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
     window: targetWindow,
   });
 
-  const badgeId = useId();
-  const childId = useId(
-    isValidElement<HTMLAttributes<HTMLElement>>(children)
-      ? children.props?.id
-      : undefined
-  );
+  let valueText = value;
 
-  let badgeContentValue = badgeContent;
-  if (badgeContentValue > max) {
-    badgeContentValue = `${max}+`;
+  if (typeof value === "number" && !max) {
+    valueText = value > 999 ? "999+" : value;
+  } else if (typeof value === "number" && max) {
+    valueText = value > max ? `${max}+` : value;
+  } else if (typeof value === "string") {
+    valueText = value.length > 4 ? `${value.slice(0, 4)}...` : value;
```

Comment:
Will this impact screen reader?

Reply:
Sorry what do you mean ? Won't the screen reader just read whatever is rendered, why would the logic have an impact  ?

Reply:
Screen reader relies on DOM information. So JS modification of value would appear to be a string with `...` instead of the full string.

Reply:
@lukeac123 you can add another span element that is visually hidden that renders the full text. Use the className `salt-visuallyHidden`. Do this only when there is truncation happening.

Reply:
And also add aria-hidden to the element that holds the truncated text. This is so that the screen reader doesn't read the text twice. So
```tsx
{truncated ? <>
  <span aria-hidden>{truncatedText}</span>
  <span className="salt-visuallyHidden">{fullText}</span>
</> : 
<span>{fullText}</span>
}
```

Reply:
conditional rendering added to render additional span with non truncated string